File: client/src/app/core/Models/order.ts
import { AddressDto } from "./address";
import { ProductResDto } from "./catalog";
import { PaymentDetails } from "./paymentDetail";


export interface OrderItem {
    id: number;
    orderId: number;
    productId: number;
    product: ProductResDto;
    quantity: number;
    totalPriceAfterDiscount: number;
    totalDiscount: number;
    totalPrice: number;
}

export interface Order{
    id: number;
    userId: number;
    orderDate: string;
    totalPriceAfterDiscount: number;
    totalDiscount: number;
    totalPrice: number;
    status: string;
    orderItems: OrderItem[];
}

export interface OrderDetailDTO {
    order: Order;
    paymentDetails: PaymentDetails;
    shippingAddress:AddressDto;
}

File: server/Entities/Order.cs

namespace server.Entities
{
    public class Order : AuditBaseEntity
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public DateTime OrderDate { get; set; }
        public decimal TotalPriceAfterDiscount { get; set; }
        public decimal TotalDiscount { get; set; }
        public decimal TotalPrice { get; set; }
        public string Status { get; set; } // e.g., Pending, Completed, Cancelled
        public ICollection<OrderItem> OrderItems { get; set; }
    }
}